### ISSUE TITLE
Error in make for c++

### ISSUE THREAD

**Issue by lyysl** [DrTimothyAldenDavis/SuiteSparse#327]
I have errors in `make`

```
CMake Error at cmake_modules/FindGMP.cmake:73 (file):
  file STRINGS file
  "/home/simon/SuiteSparse/SuiteSparse-7.0.1/SPEX/GMP_INCLUDE_DIR-NOTFOUND/gmp.h"
  cannot be read.
Call Stack (most recent call first):
  CMakeLists.txt:52 (find_package)


CMake Error at cmake_modules/FindGMP.cmake:75 (file):
  file STRINGS file
  "/home/simon/SuiteSparse/SuiteSparse-7.0.1/SPEX/GMP_INCLUDE_DIR-NOTFOUND/gmp.h"
  cannot be read.
Call Stack (most recent call first):
  CMakeLists.txt:52 (find_package)


CMake Error at cmake_modules/FindGMP.cmake:77 (file):
  file STRINGS file
  "/home/simon/SuiteSparse/SuiteSparse-7.0.1/SPEX/GMP_INCLUDE_DIR-NOTFOUND/gmp.h"
  cannot be read.
Call Stack (most recent call first):
  CMakeLists.txt:52 (find_package)


-- major:
-- minor:
-- patch:
CMake Error at cmake_modules/FindGMP.cmake:89 (string):
  string sub-command REGEX, mode MATCH needs at least 5 arguments total to
  command.
Call Stack (most recent call first):
  CMakeLists.txt:52 (find_package)


CMake Error at cmake_modules/FindGMP.cmake:90 (string):
  string sub-command REGEX, mode MATCH needs at least 5 arguments total to
  command.
Call Stack (most recent call first):
  CMakeLists.txt:52 (find_package)


CMake Error at cmake_modules/FindGMP.cmake:91 (string):
  string sub-command REGEX, mode MATCH needs at least 5 arguments total to
  command.
Call Stack (most recent call first):
  CMakeLists.txt:52 (find_package)


CMake Error at /usr/local/share/cmake-3.26/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find GMP: Found unsuitable version "..", but required is at least
  "6.1.2" (found
  /home/simon/SuiteSparse/SuiteSparse-7.0.1/SPEX/GMP_LIBRARY-NOTFOUND)
Call Stack (most recent call first):
  /usr/local/share/cmake-3.26/Modules/FindPackageHandleStandardArgs.cmake:598 (_FPHSA_FAILURE_MESSAGE)
  cmake_modules/FindGMP.cmake:103 (find_package_handle_standard_args)
  CMakeLists.txt:52 (find_package)


-- Configuring incomplete, errors occurred!
```

I tried to install gmp library but still get this error.
Thank you.

**Comment by DrTimothyAldenDavis** [comment]
The underlying problem is that GMP and MPFR are not installed, or not where cmake can find them.  

If the GMP library is not found, then the script should be more clear, so that could be fixed.  The error won't go away but the error message should be more clear.

There is an update to FindGMP.cmake and FindMPFR.cmake in the latest update to the dev branch.  Can you give that a try?

**Comment by DrTimothyAldenDavis** [comment]
I've revised FindGMP and FindMPFR.cmake so that they should give a better error message if GMP and MPFR are not found.  This is in the latest version in the dev and dev2 branches.

**Comment by DrTimothyAldenDavis** [comment]
Fixed in v7.1.0
